### PR TITLE
Remove AnalysisMetaData dependencies where not all properties are used

### PR DESCRIPTION
**Pull request by mfussenegger**
See commit messages

### CODE REVIEW

**Comment by mfussenegger** [comment]
@matriv please review

File: sql/src/main/java/io/crate/analyze/EvaluatingNormalizer.java [hunk 1]
```diff
@@ -62,43 +62,33 @@
...
+        return new EvaluatingNormalizer(functions, RowGranularity.CLUSTER, replaceMode, null, null);
+    }
 
     /**
      * @param functions         function resolver
      * @param granularity       the maximum row granularity the normalizer should try to normalize
      * @param referenceResolver reference resolver which is used to resolve paths
      * @param fieldResolver     optional field resolver to resolve fields
-     * @param inPlace           defines if symbols like functions can be changed inplace instead of being copied when changed
+     * @param replaceMode       defines if symbols like functions can be mutated or if they have to be copied
```
Comment (matriv) [inline sql/src/main/java/io/crate/analyze/EvaluatingNormalizer.java]: move 2 lines up to match the ordering of the arguments

File: sql/src/main/java/io/crate/operation/collect/sources/NodeStatsCollectSource.java [hunk 2]
```diff
@@ -94,8 +91,8 @@ public NodeStatsCollectSource(TransportActionProvider transportActionProvider,
     }
 
     @Nullable
-    protected static Collection<DiscoveryNode> nodeIds(WhereClause whereClause, Collection<DiscoveryNode> nodes,
-                                                       Functions functions) {
+    static Collection<DiscoveryNode> nodeIds(WhereClause whereClause, Collection<DiscoveryNode> nodes,
```
Comment (matriv) [inline sql/src/main/java/io/crate/operation/collect/sources/NodeStatsCollectSource.java]: Move nodes arg to a new line

**Comment by mfussenegger** [comment]
@seut want to take a look as well?

**Pull request merged**